torch/csrc/jit/frontend/function_schema_parser.cpp

```diff
@@ -27,8 +27,8 @@ namespace jit {
 
 namespace {
 struct SchemaParser {
-  SchemaParser(const std::string& str)
-      : L(std::make_shared<SourceView>(c10::string_view(str))),
+  explicit SchemaParser(const std::string& str)
+      : L(std::make_shared<Source>(c10::string_view(str))),
```

Comment:
Does this mean we are copying entire string (within Source) when schema parser is created?

Reply:
No. Source created with string_view don't copy (equiv to old SourceView) and Source created with std::string copies (equiv to old Source)

Reply:
In that case, `source` may or may not own the underlying string and that is dangerous. Could we preserve the original inheritance structure so that it is clear whether ownership is needed or not?